component/prometheus/source/api/api.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component"
+	fnet "github.com/grafana/agent/component/common/net"
+	agentprom "github.com/grafana/agent/component/prometheus"
+	"github.com/grafana/agent/pkg/util"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/storage/remote"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "prometheus.source.api",
```

Comment:
We usually use multiple dots to namespace components (some [conventions](https://github.com/grafana/agent/issues/2059) from back when we were first naming components).

I don't think we'll introduce more `prometheus.source` components, so wdyt about reconsidering the name? Throwing some ideas out there, but I'd like to hear your suggestions

* prometheus.http
* prometheus.httpreceiver
* prometheus.remotewritereceiver